plugins/signature_provider_plugin/CMakeLists.txt

```diff
@@ -0,0 +1,10 @@
+file(GLOB HEADERS "include/eosio/signature_provider_plugin/*.hpp")
+add_library( signature_provider_plugin
+             signature_provider_plugin.cpp
+             ${HEADERS} )
+
+target_link_libraries( signature_provider_plugin appbase fc http_client_plugin )
+target_include_directories( signature_provider_plugin PUBLIC include )
```

Comment:
Shouldn't this be PRIVATE and not PUBLIC.

Reply:
I think it needs to be PUBLIC because, for example, producer_plugin that uses this has to `#include <eosio/signature_provider_plugin/signature_provider_plugin.hpp>`

Reply:
Yeah, you are right. Sorry knee jerk.  I've been looking at the EOS code base over the last week and looking at all of the places of the over use of PUBLIC.  It's the reason why we can't simply use CMake PACKAGE without it barfing all of the libs and includes everywhere.

Reply:
It is one of the reasons, let me put it that way.